### ISSUE TITLE
[2019 CVPR Oral] Do Better ImageNet Models Transfer Better?

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#75]
저자들은 12개의 dataset과 16개의 모델에서 evaluation을 통해 ImageNet pre-trained model이 transfer에 영향을 주는지 실험하였으며, 결론만 정리하면 다음과 같음

1. ImageNet 성능이 좋으면 transfer learning 성능도 좋음 (linear probing, fine-tuning 둘다)
2. ImageNer에서 성능을 높이기 위해 사용한 regularizer는 transfer learning 성능에 악영향을 줌 (linear probing만 해당)
3. Task에 상관없이 architecture는 transfer가 되지만 weight는 아님 (좋은 모델은 효과가 있으나 ImageNet pre-trained weight는 효과를 못볼 수도 있음)

참고로 여기서 말하는 linear probing은 편의를 위해서 사용한 용어이지, 실제로는 feature backbone을 freeze하고 logistic regression classifier를 이용함
(not linear classifier)

저자들이 oral 세션에서 발표를 한 순서를 토대로 논문을 정리

## 1. Evaluation
<img width="800" alt="image" src="https://user-images.githubusercontent.com/49643709/173060302-f9658583-07d5-4f2c-aee9-be5095edd8ab.png">
<img width="1200" alt="image" src="https://user-images.githubusercontent.com/49643709/173060440-39857baa-c2b7-4b17-b909-de112f91af07.png">

## 2. Metric
<img width="500" alt="image" src="https://user-images.githubusercontent.com/49643709/173054526-b84c2f9c-31d9-48e1-8092-829893ef21b5.png">
<img width="500" alt="image" src="https://user-images.githubusercontent.com/49643709/173055016-e7360200-a503-431f-915f-11d7ab1bc94d.png">

결국 보고자하는 관계는 ImageNet performance와 transfer performance의 관계임

Performance를 측정하는 가장 간단한 방법으로는 acc를 보는 것인데, transfer performance를 구하는데 있어 단순히 평균을 내면 그 의미를 제대로 담지 못함

저자들은 다음과 같은 방법을 통해 transfer performance인 average accuracy를 구함

1. Log odds를 사용하여 logit-transformed accuracy를 구함 (acc 50% acc 99%에서의 1%의 차이는 다른 것을 반영하기 위해)
2. Dataset에 대하여 다양한 모델들의 logit-transformed accuracy를 이용하여 adjusted accuracy의 mean, std를 구함 (dataset의 difficulty로 인한 variance를 제거하기 위해)

### 3.1. Public Checkpoints
<img width="1200" alt="image" src="https://user-images.githubusercontent.com/49643709/173060111-05aa4fd8-8459-4557-a2aa-5c85b501ca52.png">

Public checkpoints를 이용하여 ImageNet performance와 transfer performance간의 spearman's rank correlation coefficient를 구해보면 0.37로 관계가 없다는 것을 확인할 수 있음

여기서 주목할만한 점은 ResNet과 DenseNet의 경우 ImageNet performance는 뛰어나지 않았지만 transfer performance가 매우 뛰어났다는 것임

### 3.2. Regularization
<img width="500" alt="image" src="https://user-images.githubusercontent.com/49643709/173063342-838fb5ba-6ccb-45dd-8830-c59de6adb6b2.png">

저자들은 실험을 통해 training setting이 transfer performance에 영향을 미친다는 것을 알게 됨
(Inception settings : -BN Scale, +Label Smooth, +Dropout, ResNet settings : +BN Scale, -Label Smooth, -Dropout, -Aux Head)

ImageNet을 학습하는데 있어 성능을 올리기 위해 regularization을 사용했지만, 이는 transfer performance를 저하시킨다는 것

저자들은 이를 통해 public checkpoint를 사용하는 것이 아닌 train setting을 통일하여 ImageNet에서 모델을 학습함

### 3.3. How to Train on ImageNet
<img width="1500" alt="image" src="https://user-images.githubusercontent.com/49643709/173065143-d51fda56-c873-496f-af8b-0adde76d86d7.png">
<img width="1500" alt="image" src="https://user-images.githubusercontent.com/49643709/173065046-b85346c8-1b0d-486b-9a62-74a5da021dab.png">

Regularization technique와 learning rate, scheduler과 같은 hyperparameter를 고정해놓고 model retrain

똑같은 방식으로 학습해도 비슷하게 성능이 다 나오기 때문에 그냥 통일해서 사용
(성능 gap이 있는 모델들은 regularization을 떼서 그렇다는 것을 실험을 통해 확인했다고 함)

### 4.1. Linear probing
<img width="874" alt="image" src="https://user-images.githubusercontent.com/49643709/173111228-390485c3-88ef-46d7-8b6e-3f27b563479a.png">
<img width="405" alt="image" src="https://user-images.githubusercontent.com/49643709/173111140-2ac09623-e2f0-4054-bd50-560f1629e8ed.png">

ImageNet pre-trained model을 freeze하고 logistic regression classifier 학습

ImageNet에서 성능이 좋은 모델은 linear probing 성능도 좋음

대신 regularization을 신중하게 써야함

### 4.2. Fine-tune
<img width="876" alt="image" src="https://user-images.githubusercontent.com/49643709/173111292-67f694c5-1587-4e9f-b51f-21b88105af05.png">
<img width="477" alt="image" src="https://user-images.githubusercontent.com/49643709/173112020-5fc71e39-8263-4b02-8af9-3cfb6bdc8e87.png">

ImageNet pre-trained model을 fine-tune

ImageNet에서 성능이 좋은 모델은 fine-tune 성능도 좋음

Linear probing과 다르게 regularization에 크게 영향받지 않음
(+BN scale, -Label Smooth 효과 있음, -Dropout, -Aux Head는 fine-tune할 때 사용하면 가끔 효과 있음)

### 4.3. Random initialization
<img width="863" alt="image" src="https://user-images.githubusercontent.com/49643709/173111337-90821d8b-f276-4482-bc41-d50bf19c2d48.png">

저자들은 correlation이 낮아 분석을 진행했는데 결과는 다음과 같음

- Dataset 수가 10,000개가 안되는 경우, correlation이 낮아 연관성이 있다고 보기 힘듬 (r=0.29)
- Larger dataset에서는 correlation이 높아 연관성이 있음 (r=0.86)

종합적으로 보면 경향성이 약하지만 dataset이 어느정도 있다고 하면 ImageNet에서 성능이 좋았던 모델이 transfer dataset에서도 성능이 좋았다고 말할 수 있음

### 4.4. Benefits of better models are comparable to specialized methods for transfer learning
<img width="936" alt="image" src="https://user-images.githubusercontent.com/49643709/173114215-b38bb137-14a0-416b-943c-5a331a106e6b.png">

Simply using a better model can yield gains comparable to specialized techniques

### 4.5. ImageNet pretraining does not necessarily improve accuracy on fine-grained tasks
<img width="930" alt="image" src="https://user-images.githubusercontent.com/49643709/173114392-344dad19-032c-4880-9037-0da92327b50b.png">
<img width="468" alt="image" src="https://user-images.githubusercontent.com/49643709/173114438-fd7fa61a-2a6f-470f-bca1-7b1d1caab0ae.png">

대부분의 경우 fine-tune이 linear probing보다 결과가 좋으며, dataset이 클수록 차이가 커짐

대부분의 경우 fine-tune이 random initialization보다 결과가 좋으며, dataset이 클수록 차이가 줄어듬

하지만 fine-grained classification에서는 fine-tune과 random initialization의 차이가 미미함

예시로 Stanford Cars dataset과 FGVC Aircraft dataset이 있는데, 해당 dataset의 class 일부는 ImageNet에 있음

그리고 두 데이터셋 모두 data 수가 적음에도 fine-tune, random initialization 성능이 비슷함

즉 ImageNet으로 배운 representation이 다른 natural image task에 adapt가 잘 안된다는 것임
(it is surprising that common transfer learning approaches cannot always profitably adapt features learned from a large natural-image to a much smaller natural image dataset)

### 4.6. ImageNet pretraining accelerates convergence
<img width="931" alt="image" src="https://user-images.githubusercontent.com/49643709/173117221-dfa90eba-1287-4798-b0f4-00d72d8c3665.png">

그럼 저렇게 adapt가 안되는 경우 ImageNet pretraining이 필요 없지 않느냐?라고 할 수 있음

그에 대해서 저자들은 ImageNet pretraining을 하고 fine-tune을 하면 fine-tune 시에 학습 시간이 줄어든다는 장점이 있다는 것을 실험을 통해 보임

### 4.7. Accuracy benefits of ImageNet pretraining fade quickly with dataset size
<img width="932" alt="image" src="https://user-images.githubusercontent.com/49643709/173117433-c598a481-1646-4bb6-9d95-0c9fd20782a9.png">

- Data가 아주 sparse한 경우 : linear probing이 효과가 가장 좋음
- Data가 적당히 있는 경우 : fine-tune이 효과가 가장 좋음
- Data가 많은 경우 : random initialization 성능이 fine-tune과 비슷해짐
